ESP8266_Influxdb 3.12.1 error 'virtual int InfluxDBClient::BatchStreamer::availableForWrite()' marked 'override', but does not override

__Steps to reproduce::

1. Using sloeber I created a simplke arduino project for my ESP32 FireBeetle.
2. I added following code  as my *.ino file:

`/**
 * Basic Write Example code for InfluxDBClient library for Arduino
 * Data can be immediately seen in a InfluxDB UI: wifi_status measurement
 * Enter WiFi and InfluxDB parameters below
 *
 * Measures signal level of the actually connected WiFi network
 * This example supports only InfluxDB running from unsecure (http://...)
 * For secure (https://...) or Influx Cloud 2 use SecureWrite example
 **/

#include <WiFiMulti.h>
#include <InfluxDbClient.h>

WiFiMulti wifiMulti;

#define DEVICE "ESP32"
#define WIFI_SSID "mySSID"
#define WIFI_PASSWORD "mypassword"
#define INFLUXDB_URL "192.168.178:49"
#define INFLUXDB_TOKEN "WRITE_YOUR_API_TOKEN"
#define INFLUXDB_ORG "WRITE_YOUR_ORG"
#define INFLUXDB_BUCKET "WRITE_YOUR_BUCKET"


// InfluxDB client instance
InfluxDBClient client(INFLUXDB_URL, INFLUXDB_ORG, INFLUXDB_BUCKET, INFLUXDB_TOKEN);
// InfluxDB client instance for InfluxDB 1
//InfluxDBClient client(INFLUXDB_URL, INFLUXDB_DB_NAME);

// Data point
Point sensor("wifi_status");

void setup() {
  Serial.begin(115200);

  // Connect WiFi
  Serial.println("Connecting to WiFi");
  WiFi.mode(WIFI_STA);
  wifiMulti.addAP(WIFI_SSID, WIFI_PASSWORD);
  while (wifiMulti.run() != WL_CONNECTED) {
    Serial.print(".");
    delay(500);
  }
  Serial.println();

  // Set InfluxDB 1 authentication params
  //client.setConnectionParamsV1(INFLUXDB_URL, INFLUXDB_DB_NAME, INFLUXDB_USER, INFLUXDB_PASSWORD);

  // Add constant tags - only once
  sensor.addTag("device", DEVICE);
  sensor.addTag("SSID", WiFi.SSID());

  // Check server connection
  if (client.validateConnection()) {
    Serial.print("Connected to InfluxDB: ");
    Serial.println(client.getServerUrl());
  } else {
    Serial.print("InfluxDB connection failed: ");
    Serial.println(client.getLastErrorMessage());
  }
}

void loop() {
  // Store measured value into point
  sensor.clearFields();
  // Report RSSI of currently connected network
  sensor.addField("rssi", WiFi.RSSI());
  // Print what are we exactly writing
  Serial.print("Writing: ");
  Serial.println(client.pointToLineProtocol(sensor));
  // If no Wifi signal, try to reconnect it
  if (wifiMulti.run() != WL_CONNECTED) {
    Serial.println("Wifi connection lost");
  }
  // Write point
  if (!client.writePoint(sensor)) {
    Serial.print("InfluxDB write failed: ");
    Serial.println(client.getLastErrorMessage());
  }

  //Wait 10s
  Serial.println("Wait 10s");
  delay(10000);
}
`
__Expected behavior:__
I expect to compile without errors

__Actual behavior:__
I get this compile error:

> C:\Sloeber\arduinoPlugin\libraries\ESP8266_Influxdb\3.12.1\src\InfluxDbClient.h:211:21: error: 'virtual int InfluxDBClient::BatchStreamer::availableForWrite()' marked 'override', but does not override virtual int availableForWrite() override;


__Specifications:__
 - Library Version: 3.12.1
 - InfluxDB Version: latest
 - Board/ESP chip: ESP32 FireBeetle
 - Device Arduino SDK version: Sloeber

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,8 @@
 # Changelog
+##  unreleased
+### Fixes
+- [200](https://github.com/tobiasschuerg/InfluxDB-Client-for-Arduino/pull/200) - Backward compatible compilation. Solves _marked 'override', but does not override_ errors.
+
 ##  3.12.2 [2022-09-30]
 ### Fixes
 - [198](https://github.com/tobiasschuerg/InfluxDB-Client-for-Arduino/pull/198) - Effective passing Point by value

--- a/src/InfluxDbClient.cpp
+++ b/src/InfluxDbClient.cpp
@@ -578,7 +578,7 @@ int InfluxDBClient::postData(Batch *batch) {
 
     BatchStreamer *bs = new BatchStreamer(batch);
     INFLUXDB_CLIENT_DEBUG("[D] Writing to %s\n", _writeUrl.c_str());
-    INFLUXDB_CLIENT_DEBUG("[D] Sending:\n");       
+    INFLUXDB_CLIENT_DEBUG("[D] Sending %d:\n", bs->available());       
     
     if(!_service->doPOST(_writeUrl.c_str(), bs, PSTR("text/plain"), 204, nullptr)) {
         INFLUXDB_CLIENT_DEBUG("[D] error %d: %s\n", _service->getLastStatusCode(), _service->getLastErrorMessage().c_str());
@@ -724,15 +724,23 @@ int InfluxDBClient::BatchStreamer::availableForWrite() {
     return 0;
 }
 
-#if defined(ESP8266)        
+void InfluxDBClient::BatchStreamer::reset() {
+    _read = 0;
+    _pointer = 0;
+    _linePointer = 0;
+}
+
 int InfluxDBClient::BatchStreamer::read(uint8_t* buffer, size_t len) {
-    INFLUXDB_CLIENT_DEBUG("BatchStream::read %d\n", len);
+    INFLUXDB_CLIENT_DEBUG("BatchStream::read\n");
     return readBytes((char *)buffer, len);
 }
-#endif
-size_t InfluxDBClient::BatchStreamer::readBytes(char* buffer, size_t len) {
 
-    INFLUXDB_CLIENT_DEBUG("BatchStream::readBytes %d\n", len);
+size_t InfluxDBClient::BatchStreamer::readBytes(char* buffer, size_t len) {
+#if defined(ESP8266)
+    INFLUXDB_CLIENT_DEBUG("BatchStream::readBytes %d, free_heap %d, max_alloc_heap %d, heap_fragmentation  %d\n", len, ESP.getFreeHeap(), ESP.getMaxFreeBlockSize(), ESP.getHeapFragmentation());
+#elif defined(ESP32)
+    INFLUXDB_CLIENT_DEBUG("BatchStream::readBytes %d, free_heap %d, max_alloc_heap %d\n", len, ESP.getFreeHeap(), ESP.getMaxAllocHeap());
+#endif
     unsigned int r=0;
     for(unsigned int i=0;i<len;i++) {
         if(available()) {

--- a/src/InfluxDbClient.h
+++ b/src/InfluxDbClient.h
@@ -204,17 +204,17 @@ class InfluxDBClient {
       public:
         BatchStreamer(Batch *batch) ;
         virtual ~BatchStreamer() {};
+        // Clears pointers to start reading from beginning
+        void reset();
 
           // Stream overrides
         virtual int available() override;
 
-        virtual int availableForWrite() override;
+        virtual int availableForWrite();
 
         virtual int read() override;
-#if defined(ESP8266)        
-        virtual int read(uint8_t* buffer, size_t len) override;
-#endif
-        virtual size_t readBytes(char* buffer, size_t len) override;
+        virtual int read(uint8_t* buffer, size_t len);
+        virtual size_t readBytes(char* buffer, size_t len);
 
         virtual void flush() override {};
         virtual int peek()  override;

--- a/test/Test.cpp
+++ b/test/Test.cpp
@@ -50,7 +50,6 @@ void Test::run() {
     testFluxTypes();
     testFluxTypesSerialization();
     testTimestampAdjustment();
-
     testFluxParserEmpty();
     testFluxParserSingleTable();
     testFluxParserNilValue();
@@ -74,7 +73,6 @@ void Test::run() {
     testLargeBatch();  
     testFailedWrites();
     testTimestamp();
-    
     testRetryOnFailedConnection();
     testRetryOnFailedConnectionWithFlush();
     testNonRetry();
@@ -430,6 +428,18 @@ void Test::testBatch() {
         TEST_ASSERT(buff[i++] == line[j]);
     }
     TEST_ASSERT(buff[i++] == '\n');
+    buff[0] = 0;
+    str.reset();
+    //Test Stream API
+    Stream *s  = &str;
+    TEST_ASSERT(s->available() == len*2+2);
+    TEST_ASSERT(s->readBytes(buff, len+1) == len+1);
+    TEST_ASSERT(s->available() == len+1);
+    TEST_ASSERT(s->readBytes(buff+len+1, len) == len);
+    TEST_ASSERT(s->available() == 1);
+    TEST_ASSERT(s->readBytes(buff+2*len+1, 1) == 1);
+    TEST_ASSERT(s->available() == 0);
+
     delete [] buff;
     TEST_END();
 }
@@ -2459,7 +2469,8 @@ void Test::testLargeBatch() {
 #if defined(ESP8266)
     int batchSize = 330;
 #elif defined(ESP32)
-    int batchSize = 2047;
+    // 2.0.4. introduces a memory hog which causes original 2048 lines cannot be sent
+    int batchSize = 1950;
 #endif
     int len =strlen(line); 
     int points = free/len;
@@ -2477,6 +2488,7 @@ void Test::testLargeBatch() {
             WS_DEBUG_RAM("Full batch");
         }
         TEST_ASSERTM(client.writeRecord(line),client.getLastErrorMessage());
+        yield();
     }
     WS_DEBUG_RAM("Data sent");
     String query = "select";

